packages/trie/src/trie/trie.ts

```diff
@@ -59,8 +60,12 @@ export class Trie {
     this.EMPTY_TRIE_ROOT = this.hash(RLP_EMPTY_STRING)
     this._hashLen = this.EMPTY_TRIE_ROOT.length
     this._root = this.EMPTY_TRIE_ROOT
+    if (opts?.persistRoot === true) {
+      this._deleteFromDB = true
```

Comment:
Yes, in the past if you deleted keys (so `Trie.del`) this would also delete the keys from the DB. However, this means you cannot rollback into past tries, since you now have deleted tries. This lead to us not being able to handle reorgs in the client / tests. This is essentially the non-active version of trie pruning, since in active trie pruning you actively delete all nodes which cannot be reachable, but in the `deleteFromDb` it only deletes the node.

Reply:
I have no super strong opinions here, someone else has to chime in here. The `deleteFromDB` is essentially the pruned-"lite" version, so I think it might make some sense to remove/deprecate it in favor of this new pruning?

Reply:
@faustbrian what name would you suggest for this option? (We can tackle this in another PR)

Reply:
Also wondering if we actually need to keep deleteFromDB.

Reply:
I will remove it, it indeed does not make much sense to have two modes.